cf_xarray/accessor.py

```diff
@@ -79,6 +79,10 @@ def apply_mapper(
     It should return a list in all other cases including when there are no
     results for a good key.
     """
+
+    if not isinstance(key, str) and default is not None:
+        return default
```

Comment:
Should we add and `error` if key is not a string and default is not None? Although apply mapper is mainly used internally, so the error wouldn't mean much to users?

Reply:
Yes let's raise an error, it will help when it's raised :)

Something like this?

``` python
if not isinstance(key, str):
    if default is None
        raise
    else:
		return default
```